modules/dnn/src/layers/trilu_layer.cpp

```diff
@@ -42,18 +41,19 @@ class TriluLayerImpl CV_FINAL : public TriluLayer {
             for (int i = 0; i < cdims; ++i)
                 loops *= shape_input[i];
 
-            float *dst = outputs[0].ptr<float>();
+            const size_t elemSize = outputs[0].elemSize();
+            uchar *dst = outputs[0].ptr<uchar>();
             auto fn = [&](const Range &r) {
-
                 for (int i = r.start; i < r.end; i++) {
                     for(int l=0; l < m; l+=1) {
                         int cmin = upperTri ? 0 : (l + k + 1);
                         cmin = std::max(cmin, 0);
                         const int cmax = upperTri ? min(l + k -1, w-1) : w-1;
                         const int num_zeros = cmax - cmin + 1;
-                        auto *cur_dst = dst + ((w * h) * i + (w * l + cmin));
+                        size_t offset = (static_cast<size_t>(w) * h * i + (static_cast<size_t>(w) * l + cmin)) * elemSize;
+                        auto *cur_dst = dst + offset;
                         if (cmin < w && num_zeros > 0)
-                            std::memset(cur_dst, 0, sizeof(float) * num_zeros);
+                            std::memset(cur_dst, 0, elemSize * num_zeros);
```

Comment:
just wondering if it will be a good idea to assert `elemSize` is positive or check in `if (cmin < w && num_zeros > 0 && elemSize > 0)` . Please feel free to ignore if it is not applicable